nixos/modules/services/misc/owncast.nix

```diff
@@ -0,0 +1,101 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let cfg = config.services.owncast;
+in {
+
+  options.services.owncast = {
+
+    enable = mkEnableOption "owncast";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/owncast";
+      description = "The directory where owncast stores its data files.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for owncast.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "User account under which owncast runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "Group under which owncast runs.";
+    };
+
+    listen = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "127.0.0.1";
+      description = "The IP address to bind owncast to.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 80;
+      description = ''
+        TCP port where owncast web-gui listens.
+      '';
+    };
+
+    rtmp-port = mkOption {
+      type = types.port;
+      default = 1935;
+      description = ''
+        TCP port where owncast rtmp service listens.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.tmpfiles.rules = [
+      "L+ '${cfg.dataDir}/static' - - - - ${pkgs.owncast.src}/static"
+      "C '${cfg.dataDir}/webroot' 0700 - - - ${pkgs.owncast.src}/webroot"
+    ];
+
+    systemd.services.owncast = {
+      wantedBy = [ "default.target" ];
+
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        WorkingDirectory = cfg.dataDir;
+        StateDirectory = baseNameOf cfg.dataDir;
```

Comment:
What if `dataDir` isn't under `/var/lib`? This wouldn't work.

Reply:
Looks like you might have been better off with the `preStart` after all... Sorry about that. Take a look at [this](https://github.com/NixOS/nixpkgs/commit/2e3220d65faaaa236a6ca1916f9ba95d6467fa80#diff-9d39641ee77d01c796971694d01fa548860751b9055db1890f3a680a59f06fe4R13-R15) and [this](https://github.com/NixOS/nixpkgs/commit/2e3220d65faaaa236a6ca1916f9ba95d6467fa80#diff-9d39641ee77d01c796971694d01fa548860751b9055db1890f3a680a59f06fe4R72-R83).